### PR TITLE
implemented enabling / disabling child agencies

### PR DESCRIPTION
**Pull request by DFSko**
implemented enabling / disabling child agencies
resolves happy-travel/agent-app-project#4

### CODE REVIEW

**Comment by DFSko** [comment]
Need to check the module "pg_trim" is installed in PostgreSQL before merging

File: Api/Services/Agents/AgencyManagementService.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+
+        public Task<Result> ActivateChildAgency(int agencyId, AgentContext agent)
+            => GetAgency(agencyId, agent)
+                .Bind(agency => ChangeActivityStatus(agency, ActivityStatus.Active));
+
+        
+        private async Task<Result<Agency>> GetAgency(int agencyId, AgentContext agent)
+        {
+            var agency = await _context.Agencies.FirstOrDefaultAsync(ag => ag.Id == agencyId && ag.ParentId == agent.AgencyId);
```
Comment (mingazhev) [inline Api/Services/Agents/AgencyManagementService.cs:35]: Why not single or default?

File: Api/Services/Agents/AgencyManagementService.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                foreach (var account in agencyAccounts)
+                    account.IsActive = convertedStatus;
+
+                _context.UpdateRange(agencyAccounts);
+                await _context.SaveChangesAsync();
+            }
+        }
+        
+        
+        private static bool ConvertToDbStatus(ActivityStatus status) => status == ActivityStatus.Active;
```
Comment (mingazhev) [inline Api/Services/Agents/AgencyManagementService.cs:76]: Arrow to the next line

File: Api/Services/Agents/AgencyManagementService.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Agents/AgencyManagementService.cs]: It's copy-paste from AdminAgencyManagementService. Fixed

File: Api/Services/Agents/AgencyManagementService.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Agents/AgencyManagementService.cs]: Fixed

Comment (kirillta) [inline Api/Services/Agents/AgencyManagementService.cs]: Why always `active`?

File: Api/Services/Agents/AgencyManagementService.cs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                foreach (var account in agencyAccounts)
+                    account.IsActive = convertedStatus;
+
+                _context.UpdateRange(agencyAccounts);
+                await _context.SaveChangesAsync();
+            }
+        }
+        
+        
+        private static bool ConvertToDbStatus(ActivityStatus status)
```
Comment (kirillta) [inline Api/Services/Agents/AgencyManagementService.cs:76]: Why always `active`?

Comment (DFSko) [reply Api/Services/Agents/AgencyManagementService.cs:76]: Not always. This method converts enum ActivityStatus to boolean
ActivityStatus.Active => true
ActivityStatus.NotActive => false

Comment (DFSko) [reply Api/Services/Agents/AgencyManagementService.cs]: Implemented through switch

**Review by mingazhev — APPROVED** [review]

**Pull request merged**